padertorch/contrib/examples/vae_mnist/mnist_vae_example.py

```diff
@@ -0,0 +1,195 @@
+
+        MSE = self.MSE_loss(prediction, images)  # reconstruction error
+        # Kullback-Leibler divergence
+        KLD = -0.5 * torch.sum(1 + outputs['std'] - outputs['mean'].pow(2) - outputs['std'].exp())
+
+        loss = MSE + KLD
+
+        return dict(
+            loss=loss,
+        )
+
+
+class VAE(nn.Module):
+    """ A simple Variational Autoencoder"""
+
+    def __init__(self):
+        super(VAE, self).__init__()
+        self.kernel_size = 4
+
+        # encoder
+        self.conv_1 = torch.nn.Conv2d(in_channels=1, out_channels=64, kernel_size=self.kernel_size,
+                                      stride=2)
+        self.conv_2 = torch.nn.Conv2d(in_channels=64, out_channels=64, kernel_size=self.kernel_size,
+                                      stride=2)
+        self.conv_3 = torch.nn.Conv2d(in_channels=64, out_channels=1, kernel_size=self.kernel_size,
+                                      stride=1)
+        self.linear_1 = torch.nn.Linear(in_features=2*2, out_features=1)
+        self.linear_2 = torch.nn.Linear(in_features=2*2, out_features=1)
+        self.lrelu = torch.nn.LeakyReLU()
+
+        # decoder
+        self.linear_3 = torch.nn.Linear(in_features=1, out_features=2*2)
+
+        self.transconv_1 = torch.nn.ConvTranspose2d(in_channels=1, out_channels=64, kernel_size=self.kernel_size,
+                                                   stride=2)
+        self.transconv_2 = torch.nn.ConvTranspose2d(in_channels=64, out_channels=64, kernel_size=self.kernel_size,
+                                                   stride=2)
+        self.transconv_3 = torch.nn.ConvTranspose2d(in_channels=64, out_channels=1, kernel_size=self.kernel_size,
+                                                   stride=2)
+        self.relu = torch.nn.ReLU()
+
+    def encode(self, x):
+        x = self.conv_1(x)
+        x = self.lrelu(x)
+        x = self.conv_2(x)
+        x = self.lrelu(x)
+        x = self.conv_3(x)
+        x = self.lrelu(x)
+        x = rearrange(x, 'b c x y -> b (c x y)')
+        mu = self.linear_1(x)  # mean value
+        std = 0.5 * self.linear_2(x)  # standard deviation
+        eps = torch.randn_like(std)
+        return mu + eps*std, mu, std
+
+    def decode(self, x):
+        x = self.linear_3(x)
+        x = rearrange(x, 'b (x y) -> b x y', y=2, x=2)
+        x = x[:, None, :, :]
+        x = self.transconv_1(x)
+        x = self.relu(x)
+        x = self.transconv_2(x)
+        x = self.relu(x)
+        x = self.transconv_3(x)
+        return x
+
+    def forward(self, input):
+        x, mu, std = self.encode(input)
+        y = self.decode(x)
+        return y[:, :, :28, :28], mu, std
+
+
+@ex.config
+def config():
+    epochs = 2          # Use resume=True to train for more epochs
+    batch_size = 128
+    resume = False      # PT: Continue from checkpoints
+    logfilepath = os.environ['STORAGE_ROOT'] + '/mnist'  # PT creates log files
```

Comment:
you should first check whether the STORAGE_ROOT variable is set and give a more informative error message